### PR TITLE
refactor code to update data-model first

### PR DESCRIPTION
**Pull request by holladayian**
# Iteration Number
3

- [x] Are there any comments that need to be removed?
Yes. Lines 15-17 need to be removed. I just caught them upon this PR. They are now unnecessary, due to the completion of the event handling function.
- [x] Did you check all of the functionality of the webpage?
I sure did, and it works great.
- [ ] Do all parties understand every line of code?
This was a refactor done solely by Ian, however it built upon concepts we understood, and were working on as a group.
- [x] Is the code DRY/SRP
I hope so.

## Questions
Are we completely DOM-data model separated? Might be a question of a mentor.  

## Notes
- We need to go over this refactoring as a group.
- We should also talk to as many mentors as we can to see how we understand the code, and if we can refactor it in any ways.

## Problems
I hope not

## Screenshot
Screen Shot 2020-07-28 at 3 15 14 AM

@holladayian
@timkeresey
@saraho1123

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -8,29 +8,27 @@ var ideaFormSection = document.querySelector('.idea-form');
...
-//   if (event.target === deleteCardButton) {
-//     starFavorite();
-//     deleteCard();
-//   }
-// }
+function clickHandler(event) {
+  if (event.target === saveButton) {
+    createIdeaObject(event);
+  }
+  if (event.target.classList.contains("star")) {
```
Comment (saraho1123) [inline src/main.js:25]: Oh! That was brilliant! And I learned something new ... had no idea you could use contains with classList. Great to know!

**Review by saraho1123 — APPROVED** [review]
Thanks for continuing to look at this. Everything you did makes sense and looks like it was variations of things we were trying last night, but just didn't quite have right yet. Looking forward to going over it all together, but so far it looks clear and DRY to me.

File: src/idea.js [hunk 2]
```diff
@@ -12,22 +12,10 @@ class Idea {
...
-  //     starOutline.classList.remove("hidden");
-  //   }
-  //   console.log(ideaArray);
-  // }
     if (this.star) {
-      return "./assets/star-active.svg";
+      this.star = false;
     } else {
-      return "./assets/star.svg";
+      this.star = true;
```
Comment (timkeresey) [inline src/idea.js:18]: This seems clear and concise enough.

File: src/main.js [hunk 3]
```diff
@@ -119,18 +115,9 @@ function disableSaveButton() {
...
-    if (cardID == ideaArray[i].id) {
-      // starActive.classList.toggle("hidden");
-      // starOutline.classList.toggle("hidden");
-      if (ideaArray[i].star) {
-        ideaArray[i].star = false;
-      } else {
-        ideaArray[i].star = true;
-      }
+    if (Number(event.target.id) == ideaArray[i].id) {
+      ideaArray[i].updateIdea();
```
Comment (timkeresey) [inline src/main.js:120]: much simpler.

**Comment by timkeresey** [comment]
Great, thank you for refactoring. I think some mentor review would be a great idea.

**Pull request merged**